pyomo/gdp/plugins/multiple_bigm.py

```diff
@@ -0,0 +1,724 @@
+
+        obj.deactivate()
+
+    def _get_disjunct_relaxation_block(self, disjunct, transBlock):
+        if disjunct.transformation_block is not None:
+            return disjunct.transformation_block
+
+        # create a relaxation block for this disjunct
+        relaxedDisjuncts = transBlock.relaxedDisjuncts
+        relaxationBlock = relaxedDisjuncts[len(relaxedDisjuncts)]
+
+        relaxationBlock.localVarReferences = Block()
+
+        # add the map that will link back and forth between transformed
+        # constraints and their originals.
+        relaxationBlock._constraintMap = {
+            'srcConstraints': ComponentMap(),
+            'transformedConstraints': ComponentMap()
+        }
+
+        # add mappings to source disjunct (so we'll know we've relaxed)
+        disjunct._transformation_block = weakref_ref(relaxationBlock)
+        relaxationBlock._src_disjunct = weakref_ref(disjunct)
+
+        return relaxationBlock
+
+    def _transform_disjunct(self, obj, transBlock, active_disjuncts, Ms):
+        # We've already filtered out deactivated disjuncts, so we know obj is
+        # active.
+
+        # Make a relaxation block if we haven't already.
+        relaxationBlock = self._get_disjunct_relaxation_block(obj, transBlock)
+
+        # Transform everything on the disjunct
+        self._transform_block_components(obj, active_disjuncts, Ms)
+
+        # deactivate disjunct so writers can be happy
+        obj._deactivate_without_fixing_indicator()
+
+    def _transform_block_components(self, disjunct, active_disjuncts, Ms):
+        # We don't know where all the BooleanVars are used, so if there are any
+        # that logical_to_linear didn't transform, we need to do it now
+        for boolean in disjunct.component_data_objects(BooleanVar,
+                                                       descend_into=Block,
+                                                       active=None):
+            if isinstance(boolean._associated_binary,
+                          _DeprecatedImplicitAssociatedBinaryVariable):
+                parent_block = boolean.parent_block()
+                new_var = Var(domain=Binary)
+                parent_block.add_component(
+                    unique_component_name(parent_block,
+                                          boolean.local_name + "_asbinary"),
+                    new_var)
+                boolean.associate_binary_var(new_var)
+
+        # add references to all local variables on block (including the
+        # indicator_var). We won't have to do this when the writers can find
+        # Vars not in the active subtree.
+        varRefBlock = disjunct._transformation_block().localVarReferences
+        for v in disjunct.component_objects(Var, descend_into=Block,
+                                            active=None):
+            varRefBlock.add_component(unique_component_name(
+                varRefBlock, v.getname(fully_qualified=True)), Reference(v))
```

Comment:
Per #2605, shouldn’t this be `fully_qualified=False`?

Reply:
Yes...